src/utils.cpp

```diff
@@ -842,19 +852,24 @@ resolve_binary_path(const std::string &cmd, const char *env_paths, int pid)
   if (cmd.find("*") != std::string::npos)
     candidate_paths = ::expand_wildcard_paths(candidate_paths);
 
+  std::unordered_set<std::string> abs_path;
```

Comment:
Why can't we just change `valid_executable_paths` (and consequently the return type of this function) to `std::unordered_set` and avoid the copy below?

Reply:
It's wired, On my `CentOS-Stream9`, there are no failed tests at all.

```
[----------] Global test environment tear-down
[==========] 401 tests from 22 test suites ran. (610238 ms total)
[  PASSED  ] 401 tests.
```

I'll try another approach, maybe submit new V2 patch, thanks.